### PR TITLE
Feat/bumpoverride

### LINKED ISSUE #30: Support for bumpoverride
### What did you have in mind?

The new [2.1.0 version-bumper](https://github.com/TomerFi/version-bumper/releases/tag/2.1.0) includes a new feature for overriding the bump type.

This should be implemented for this action.
The dependency version was already bumped here https://github.com/TomerFi/version-bumper-action/commit/343444a64c19c3bd8035df4570c45c4c45a33624.

### Are you trying to fix a problem?

_No response_

### Any lead on how this feature may be implemented?

_No response_

### PR DESCRIPTION
**Pull request by MisterTimn**
# Pull Request

## Description

> Describe what you did and why.

**Related issue (if any):** fixes #30 

Added inputs and docs for bumpoverride.

## Checklist

- [x] I have followed this repository's contributing guidelines.
- [x] I have used the conventional commits specification for my commit messages.
- [x] I have signed my commits.
- [x] I will adhere to the project's code of conduct.

## Additional information

Not sure on the readme entry, maybe a second example using the override is in order in addendum? I wouldn't showcase it in the full usage example

### CODE REVIEW

**Comment by MisterTimn** [comment]
I had an issue with GPG yesterday but managed to fix it, so only some commits are signed. As for conventional commits, I used the style in every commit but unsure on what prefixes this repository adheres (I think its failing on chore or maybe format should be style?)

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM docker.io/tomerfi/version-bumper:2.0.4
+FROM docker.io/tomerfi/version-bumper:2.1.0
```
Comment (TomerFi) [inline Dockerfile:15]: This was already done by dependabot here https://github.com/TomerFi/version-bumper-action/commit/343444a64c19c3bd8035df4570c45c4c45a33624.

You need to rebase your branch.

File: README.md [hunk 2]
```diff
@@ -52,12 +52,13 @@ identified from [commit messages][3]:
...
-| `changelog` | Set true to create a changelog-X.md file               | `false`               |
-| `preset`    | Preset for creating the changelog                      | `conventionalcommits` |
-| `folder`    | Folder in the workspace in which the .git repo resides | `./`                  |
+| Key             | Description                                                           | Default value         |
+| --------------- | --------------------------------------------------------------------- | --------------------- |
+| `label`         | Build label for the development iteration                             | `.dev`                |
+| `changelog`     | Set true to create a changelog-X.md file                              | `false`               |
+| `preset`        | Preset for creating the changelog                                     | `conventionalcommits` |
+| `folder`        | Folder in the workspace in which the .git repo resides                | `./`                  |
+| `bumpoverride`  | Override the version bump, can be either `major`, `minor` or `patch`  |                       |
```
Comment (TomerFi) [inline README.md:61]: maybe we should add `auto` too, `auto` is the default for automatic decisions based on tags and commits.

File: action.sh [hunk 3]
```diff
@@ -12,9 +12,9 @@
 # limitations under the License.
 
 # execute script into variable
-output=$(/usr/local/scripts/entrypoint.sh --changelog $1 --preset $2 --label $3 --repopath $4)
+output=$(/usr/local/scripts/entrypoint.sh --changelog $1 --preset $2 --label $3 --repopath $4 --bumpoverride $5)
 # split output members from the variable
-read new_version next_dev_iteration <<<$(cut -f1,2 -d" " <<<$output)
+read new_version next_dev_iteration <<<"$(cut -f1,2 -d" " <<<"$output")"
```
Comment (TomerFi) [inline action.sh:17]: why is this needed?

File: action.yml [hunk 4]
```diff
@@ -13,7 +13,7 @@ inputs:
   changelog:
     description: Set true to create a changelog-X.md file (X is the version)
     required: false
-    default: false
+    default: "false"
```
Comment (TomerFi) [inline action.yml:16]: why is this needed?

File: action.yml [hunk 5]
```diff
@@ -22,6 +22,10 @@ inputs:
     description: Folder in the workspace in which the .git repo resides
     required: false
     default: ""
+  bumpoverride:
+    description: Override the version bump, can be either 'major', 'minor' or 'patch'
+    required: false
+    default: ""
```
Comment (TomerFi) [inline action.yml:28]: the default value should be `auto`, I changed from empty-string in `version-bumper` before deploying, you can see here https://github.com/TomerFi/version-bumper/commit/e67b5057850e23ba28655ae9b17976df64580b06.

**Review by TomerFi — CHANGES_REQUESTED** [review]
@MisterTimn 
Again, thank you very much for your contribution.
I think your branch needs to be rebased from the main branch.

**Comment by TomerFi** [comment]
> I had an issue with GPG yesterday but managed to fix it, so only some commits are signed. As for conventional commits, I used the style in every commit but unsure on what prefixes this repository adheres (I think its failing on chore or maybe format should be style?)

It's ok - we don't require gpg, just standard trailer sign-off,
you can add `-s` or `--signoff` when you got a `git commit` and it will add it for you.

Regarding to the style, yes you are correct, `format` is not a valid commit type, try `style` instead.

TIP: you can install and configure [commitint](https://commitlint.js.org/#/) globally, then you edit a very simple hook script then runs `commitlint` for every commit you do,  I actually use git templates to inject this hook script for all of my repos.
:-)

File: action.yml (hunk 4 above)
Comment (MisterTimn) [reply action.yml:16]: image

Auto formatted, I've disabled auto formatting in the workspace now (these commits were done before merging previous PR on tomerfi/version-bumper)

**Comment by TomerFi** [comment]
@MisterTimn 
Ignore the `Pull request build / Lint project` workflow failing, I'll take care of it.

You just fix `Auto-Me-Bot Conventional Commits` and `Auto-Me-Bot Signed Commits`,
I think because this PR is going to be squashed eventually, the fastest way for you to fix this is:

```
git reset --soft 343444a
git commit -sm "feat: added bumpoverride option to override version bumps manually"
git push origin HEAD -f
```

Once you do this - I'll fix the other failing workflow and we get this merged.

**Review by TomerFi — APPROVED** [review]

**Comment by TomerFi** [comment]
@all-contributors please add @MisterTimn for code and doc

**Comment by TomerFi** [comment]
@MisterTimn 

This is great work - thank you very much for your contribution.
Your contribution has been included in the latest [1.2.0 release](https://github.com/TomerFi/version-bumper-action/releases/tag/1.2.0).

**Pull request merged**